ledger/src/blockstore.rs

```diff
@@ -2916,6 +2916,31 @@ impl Blockstore {
         Ok(())
     }
 
+    pub fn add_transaction_status_to_batch<'a>(
```

Comment:
These two functions are almost identical except for the last write part -- even that has identical signature. Let's use a helper to share the code.

Reply:
Thx for the suggested fix. Accepted and resolved.